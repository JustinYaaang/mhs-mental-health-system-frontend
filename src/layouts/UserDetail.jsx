
import React, { Component } from 'react'
// nodejs library to set properties for components
import PropTypes from 'prop-types'
// @material-ui/core components
import withStyles from '@material-ui/core/styles/withStyles'
import InputLabel from '@material-ui/core/InputLabel'
// core components
import GridItem from 'components/Grid/GridItem.jsx'
import GridContainer from 'components/Grid/GridContainer.jsx'
import CustomInput from 'components/CustomInput/CustomInput.jsx'
import Button from 'components/CustomButtons/Button.jsx'
import Card from 'components/Card/Card.jsx'
import CardHeader from 'components/Card/CardHeader.jsx'
import CardAvatar from 'components/Card/CardAvatar.jsx'
import CardBody from 'components/Card/CardBody.jsx'
import CardFooter from 'components/Card/CardFooter.jsx'
import {fetchUserDetail } from '../services/BackendService'

import Table from "components/Table/Table.jsx";

const styles = {
  cardCategoryWhite: {
    color: 'rgba(255,255,255,.62)',
    margin: '0',
    fontSize: '14px',
    marginTop: '0',
    marginBottom: '0'
  },
  cardTitleWhite: {
    color: '#FFFFFF',
    marginTop: '0px',
    minHeight: 'auto',
    fontWeight: '300',
    fontFamily: "'Roboto', 'Helvetica', 'Arial', sans-serif",
    marginBottom: '3px',
    textDecoration: 'none'
  }
}

class UserDetail extends Component {
    constructor(props) {
        super(props);
        this.state = {
          firstname: [],
          lastname:[],
          email:[],
          postcode:[]
      }
    }

    componentWillMount() {
        const {id} = this.props.match.params;
        console.log(id)
        fetchUserDetail(id)
          .then(response => {
           
            // for (var i = 0; i < response.length; i++) {
             
            //   var d = new Date(response[i].createdAt)
            //   var dateString = d.toString()
            //   dateString = dateString.substring(0, dateString.lastIndexOf(':'))
            //     console.log(response[i].patient_id)
            //     var row = [response[i].title, response[i].patient_name, response[i].score, response[i]._id, response[i].status, dateString, response[i]._id, response[i].patient_id]
            //     if(response[i].status == 'PENDING'){
            //       rowsPending.push(row)
            //     }
            //     else{
            //       rowsResolve.push(row)
            //     }
            //   }
              console.log(response.email)
            this.setState({ firstname: response.first_name, lastname:response.last_name, email: response.email,postcode:response.postcode })
          })
          .catch(error => {
            console.error(error)
          })
      }

      render () {
        const { classes } = this.props;
        return (
          
          <div>
            <GridContainer>
              <GridItem xs={12} sm={12} md={6}>
                <Card>
                  <CardHeader color='info'>
                    <h4 className={classes.cardTitleWhite}>Patient Profile</h4>
                    <p className={classes.cardCategoryWhite}>View the detail of Patient</p>
                  </CardHeader>
                  <CardBody>
                    <GridContainer>
                      <GridItem xs={12} sm={12} md={5}>
                        <CustomInput
                          labelText='Service'
                          id='company-disabled'
                          formControlProps={{
                            fullWidth: true
                          }}
                          inputProps={{
                            disabled: true
                          }}
                        />
                      </GridItem>
                      <GridItem xs={12} sm={12} md={3}>
                        <CustomInput
                          labelText='Username'
                          id='username'
                          formControlProps={{
                            fullWidth: true
                          }}
                          inputProps={{
                              disabled: true
                            }}
                        />
                      </GridItem>
                      <GridItem xs={12} sm={12} md={4}>
                        <CustomInput
                          labelText= {this.state.email}
                          id='email-address'
                          formControlProps={{
                            fullWidth: true
                          }}
                          inputProps={{
                              disabled: true
                            }}
                        />
                      </GridItem>
                    </GridContainer>
                    <GridContainer>
                      <GridItem xs={12} sm={12} md={6}>
                        <CustomInput
                          labelText={this.state.firstname}
                          id='first-name'
                          formControlProps={{
                            fullWidth: true
                          }}
                          inputProps={{
                              disabled: true
                            }}
                        />
                      </GridItem>
                      <GridItem xs={12} sm={12} md={6}>
                        <CustomInput
                          labelText={this.state.lastname}
                          id='last-name'
                          formControlProps={{
                            fullWidth: true
                          }}
                          inputProps={{
                              disabled: true
                            }}
                        />
                      </GridItem>
                    </GridContainer>
                    <GridContainer>
                  
                      <GridItem xs={12} sm={12} md={4}>
                        <CustomInput
                          labelText={this.state.postcode}
                          id='postal-code'
                          formControlProps={{
                            fullWidth: true
                          }}
                          inputProps={{
                              disabled: true
                            }}
                        />
                      </GridItem>
                    </GridContainer>
                    <GridContainer>
                      <GridItem xs={12} sm={12} md={12}>
                      <Table
                          tableHeaderColor="info"
                          tableHead={['Questionnaire Name','Status']}
                          tableData={[
                              [ "Questionnaire 1" , "Pending"] ,
                              [ "Questionnaire 2" , "Close"] ,
                              
                          ]}
                      />
                      </GridItem>
                    </GridContainer>
                  </CardBody>
                </Card>
              </GridItem>
            </GridContainer>
            
          </div>
        )
      }
}






UserDetail.propTypes = {
  classes: PropTypes.object
}

export default withStyles(styles)(UserDetail)
