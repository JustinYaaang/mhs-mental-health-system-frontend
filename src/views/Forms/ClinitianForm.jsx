import React from 'react'
import 'assets/css/AddForm.css'
import Button from 'components/CustomButtons/Button.jsx'
import { getPersonnel, updatePersonnel, createPersonnel } from 'services/BackendService.js'
import swal from 'sweetalert2'

/**
 * Component that displays a form for a Trust Manager, Service Manager, Clinitian
 */
class ClinitianForm extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      id: this.props.id,
      hasDetails: this.props.hasDetails,
      organization: this.props.organization,
      history: this.props.history
    }
    this.onChange = this.onChange.bind(this)
    this.onSave = this.onSave.bind(this)
    console.log(this.state.hasDetails)
  }

  componentWillMount () {
    console.log(this.state.hasDetails)
    if (this.state.hasDetails) {
      console.log('in  person form' + this.state.id)
      getPersonnel(this.state.id).then(response => {
        console.log(response)
        document.getElementById('firstnameinput').value = response.first_name
        document.getElementById('lastnameinput').value = response.last_name
        document.getElementById('emailinput').value = response.email
        document.getElementById('passwordinput').value = response.password
        this.changeStep(response.role)
      })
    } else {
    }
  }

  getStep () {
    if (document.getElementById('STEP3').checked === true) return 'STEP3'
    if (document.getElementById('STEP2').checked === true) return 'STEP2'
    return undefined
  }

  changeStep (step) {
    document.getElementById(step).checked = true
  }
  onChange (event) {
  }

  onSave (event) {
    // get all the necessary details from the form

    var body = {
      id: this.state.id,
      body: {
        first_name: document.getElementById('firstnameinput').value,
        last_name: document.getElementById('lastnameinput').value,
        email: document.getElementById('emailinput').value,
        role: this.getStep()
      }
    }
    var passwordFlag = false
    // get the 2 password fields
    var pass1 = document.getElementById('passwordchange1').value
    var pass2 = document.getElementById('passwordchange2').value
    // check if the password fields contain something
    if (pass1 !== '' || pass2 !== '') {
      // if they contain and they are not the same
      console.log(pass1)
      if (pass2 !== pass1) {
        document.getElementById('passwordchange1').style.backgroundColor = '#FEC2C2'
        document.getElementById('passwordchange2').style.backgroundColor = '#FEC2C2'
        swal.fire({
          type: 'error',
          title: 'Whoops..!',
          text: 'Passwords don\'t match! '
        })

        return
      } else { // else add the password to the body
        body.body.password = pass1
        body.body.organisation_id = sessionStorage.organizationID
        passwordFlag = true
      }
    }
    // Proceed only if all fields are filled
    if (!this.allFieldsCompleted()) {
      return
    }
    if (this.state.hasDetails) { // if we are in edit mode
      // send the details and clear the password field
      updatePersonnel(body).then(response => {
        console.log(response)
        document.getElementById('passwordchange1').value = ''
        document.getElementById('passwordchange2').value = ''
        swal.fire({
          type: 'success',
          title: 'Success',
          text: 'The entry has been updated! '
        })
        this.state.history.goBack()
      })
    } else { // if we create a new person
      if (this.getStep() === undefined) {
        swal.fire({
          type: 'error',
          title: 'Error',
          text: 'Please provide the role for the clinitian(Step 2//Step 3)'
        })
        return
      }
      if (passwordFlag) {
        createPersonnel(body.body).then(response => {
          this.componentWillMount()
          console.log(response)
          swal.fire({
            type: 'success',
            title: 'Success',
            text: 'The entry has been created! '
          })
          this.state.history.goBack()
        })
      } else {
        swal.fire({
          type: 'error',
          title: 'Error',
          text: 'Password error! '
        })
      }
    }
  }

  allFieldsCompleted () {
    var a = document.getElementById('firstnameinput').value
    var b = document.getElementById('lastnameinput').value
    var c = document.getElementById('emailinput').value
    if (a === '' || b === '' || c === '') {
      swal.fire({
        type: 'error',
        title: 'Whoops',
        text: 'Please fill all the fields '
      })
      return false
    }
    return true
  }

  render () {
    return (
      <form class='trustform'>
        <label className='label-subtitle' for='label'>Please enter the details of the NHS Person</label>
        <div class='form-group'>
          <input name='email' type='email' class='form-control' id='firstnameinput' aria-describedby='emailHelp' placeholder='First Name' onChange={this.onChange} />
          <br />

          <input name='email' type='email' class='form-control' id='lastnameinput' aria-describedby='emailHelp' placeholder='Last Name' onChange={this.onChange} />
          <br />
          <input type='radio' id='STEP2' name='step' value='STEP2' /> Step 2 <br />
          <input type='radio' id='STEP3' name='step' value='STEP3' /> Step 3<br />

        </div>
        <div class='form-group'>
          <input name='email' type='email' class='form-control' id='emailinput' aria-describedby='emailHelp' placeholder='email' onChange={this.onChange} />
          <br />
          <input name='email' type='email' class='form-control' id='passwordinput' aria-describedby='emailHelp' placeholder='password' onChange={this.onChange} readOnly />
          <br />
          <input name='email' type='email' class='form-control' id='trustnameinput_postcodeinput' aria-describedby='emailHelp' placeholder='Trust Name' value={sessionStorage.organizationID} onChange={this.onChange} readOnly />
        </div>
        <br />
        <label className='label-subtitle' id='passwordlabel' for='label'>{this.state.hasDetails ? 'Change the password here:' : 'Enter password here:'}</label>
        <div>
          <input name='email' type='email' class='form-control' id='passwordchange1' aria-describedby='emailHelp' placeholder='Enter password' onChange={this.onChange} />
          <br />
          <input name='email' type='email' class='form-control' id='passwordchange2' aria-describedby='emailHelp' placeholder='Enter password again' onChange={this.onChange} />

        </div>
        <Button onClick={this.onSave} id='loginbutton' type='button' color='primary'>Save</Button>
      </form>
    )
  }
}

export default ClinitianForm

// id
// name
// address line 1
// address line 2
// post code
// description
// website link
// email
// telephone
