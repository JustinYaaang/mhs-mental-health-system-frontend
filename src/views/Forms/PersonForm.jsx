import React from 'react'
import 'assets/css/AddForm.css'
import Button from 'components/CustomButtons/Button.jsx'
import { getPersonnel, updatePersonnel, createPersonnel } from 'services/BackendService.js'
import swal from 'sweetalert2'

/**
 * Component that displays a form for a Trust Manager, Service Manager, Clinitian
 */
class PersonForm extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      id: this.props.id,
      hasDetails: this.props.hasDetails,
      organization:'trust'
    }
    this.onChange = this.onChange.bind(this)
    this.onSave = this.onSave.bind(this)
    console.log(this.state.hasDetails)
  }

  componentWillMount () {
    console.log(this.state.hasDetails)
    if (this.state.hasDetails) {
      console.log('in  person form' + this.state.id)
      getPersonnel(this.state.id).then(response => {
        console.log(response)
        document.getElementById('firstnameinput').value = response.first_name
        document.getElementById('lastnameinput').value = response.last_name
        document.getElementById('emailinput').value = response.email
        document.getElementById('passwordinput').value = response.password
        document.getElementById('trustnameinput_postcodeinput').value = response.role //! !!!
      })
    } else {
    }
  }

  onChange (event) {
  }

  onSave (event) {
    // get all the necessary details from the form
    var body = {
      id: this.state.id,
      body: {
        first_name: document.getElementById('firstnameinput').value,
        last_name: document.getElementById('lastnameinput').value,
        email: document.getElementById('emailinput').value,
        role: 'TRUSTMANAGER'

      }
    }
    var passwordFlag = false
    // get the 2 password fields
    var pass1 = document.getElementById('passwordchange1').value
    var pass2 = document.getElementById('passwordchange2').value
    // check if the password fields contain something
    if (pass1 !== '' || pass2 !== '') {
      // if they contain and they are not the same
      console.log(pass1)
      if (pass2 !== pass1) {
        document.getElementById('passwordchange1').style.backgroundColor = '#FEC2C2'
        swal.fire({
          type: 'error',
          title: 'Whoops..!',
          text: 'Passwords don\'t match! '
        })

        return
      } else { // else add the password to the body
        body.body.password = pass1
        body.body.organisation_id = sessionStorage.organizationID
        passwordFlag = true
      }
    }
    // Proceed only if all fields are filled
    if (!this.allFieldsCompleted()) {
      return
    }
    if (this.state.hasDetails) { // if we are in edit mode
      // send the details and clear the password field
      updatePersonnel(body).then(response => {
        console.log(response)
        document.getElementById('passwordchange1').value = ''
        document.getElementById('passwordchange2').value = ''
        swal.fire({
          type: 'success',
          title: 'Success',
          text: 'The entry has been updated! '
        })
      })
    } else { // if we create a new person
      if (passwordFlag) {
        createPersonnel(body.body).then(response => {
          this.componentWillMount()
          console.log(response)
          swal.fire({
            type: 'success',
            title: 'Success',
            text: 'The entry has been created! '
          })
        })
      } else {
        swal.fire({
          type: 'error',
          title: 'Error',
          text: 'Password error! '
        })
      }
    }
  }

  allFieldsCompleted () {
    var a = document.getElementById('firstnameinput').value
    var b = document.getElementById('lastnameinput').value
    var c = document.getElementById('emailinput').value
    if (a === '' || b === '' || c === '') {
      swal.fire({
        type: 'error',
        title: 'Whoops',
        text: 'Please fill all the fields '
      })
      return false
    }
    return true
  }

  render () {
    return (
      <form class='trustform'>
        <label className='label-subtitle' for='label'>Please enter the details of the NHS Person</label>
        <div class='form-group'>
          <input name='email' type='email' class='form-control' id='firstnameinput' aria-describedby='emailHelp' placeholder='First Name' onChange={this.onChange} />
          <br />

          <input name='email' type='email' class='form-control' id='lastnameinput' aria-describedby='emailHelp' placeholder='Last Name' onChange={this.onChange} />
        </div>
        <div class='form-group'>
          <input name='email' type='email' class='form-control' id='emailinput' aria-describedby='emailHelp' placeholder='email' onChange={this.onChange} />
          <br />
          <input name='email' type='email' class='form-control' id='passwordinput' aria-describedby='emailHelp' placeholder='password' onChange={this.onChange} readOnly />
          <br />
          <input name='email' type='email' class='form-control' id='trustnameinput_postcodeinput' aria-describedby='emailHelp' placeholder='Trust Name' value={sessionStorage.organizationID} onChange={this.onChange} readOnly />
        </div>
        <br />
        <label className='label-subtitle' id='passwordlabel' for='label'>{this.state.hasDetails ? 'Change the password here:' : 'Enter password here:'}</label>
        <div>
          <input name='email' type='email' class='form-control' id='passwordchange1' aria-describedby='emailHelp' placeholder='Enter password' onChange={this.onChange} />
          <br />
          <input name='email' type='email' class='form-control' id='passwordchange2' aria-describedby='emailHelp' placeholder='Enter password again' onChange={this.onChange} />

        </div>
        <Button onClick={this.onSave} id='loginbutton' type='button' color='primary'>Save</Button>
      </form>
    )
  }
}

export default PersonForm

// id
// name
// address line 1
// address line 2
// post code
// description
// website link
// email
// telephone
