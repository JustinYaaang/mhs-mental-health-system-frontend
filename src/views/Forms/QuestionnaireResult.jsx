import React, { Component } from "react";
import * as Survey from "survey-react";
import "survey-react/survey.css";
import "bootstrap/dist/css/bootstrap.css";
import "jquery-ui/themes/base/all.css";
import "nouislider/distribute/nouislider.css";
import "select2/dist/css/select2.css";
import "bootstrap-slider/dist/css/bootstrap-slider.css";
import "jquery-bar-rating/dist/themes/css-stars.css";
import $ from "jquery";
import "jquery-ui/ui/widgets/datepicker.js";
import "select2/dist/js/select2.js";
import "jquery-bar-rating";
import * as widgets from "surveyjs-widgets";
import "icheck/skins/square/blue.css";
import axios from "axios";
import { fetchQuestionnaire} from "../../services/BackendService";

import FixedActions from "components/FixedPlugin/FixedActions.jsx";

window["$"] = window["jQuery"] = $;
require("icheck");

Survey.StylesManager.applyTheme("darkblue");

widgets.icheck(Survey, $);
widgets.select2(Survey, $);
widgets.inputmask(Survey);
widgets.jquerybarrating(Survey, $);
widgets.jqueryuidatepicker(Survey, $);
widgets.nouislider(Survey);
widgets.select2tagbox(Survey, $);
widgets.signaturepad(Survey);
widgets.sortablejs(Survey);
widgets.ckeditor(Survey);
widgets.autocomplete(Survey, $);
widgets.bootstrapslider(Survey);

class QuestionnaireResult extends Component {

  constructor(props) {
    super(props);
    this.state = {
      json:
      {
        "title": "",
        "description": "",
        "completedHtml": "",
        "pages": [],
        "showProgressBar": ""
      },
      id: this.props.id, // the ID
    };
  }

  componentWillMount() {
    //const {id} = this.props.match.params;
    fetchQuestionnaire(this.state.id)
      .then(fetched_answers => {
        //this.setState({ answers: JSON.parse(fetched_answers.body) });
        var jsonData = fetched_answers.body;
        console.log(jsonData)
        var jsonFormatData = JSON.parse(jsonData);
        for (var i = 1; i < jsonFormatData.pages.length; i++) {
          if (jsonFormatData.pages[i].elements) {
            jsonFormatData.pages[0].elements = jsonFormatData.pages[0].elements.concat(jsonFormatData.pages[i].elements)
          }
        }
        jsonFormatData.pages = [jsonFormatData.pages[0]]
        jsonFormatData.pages[0].title = ""
        jsonData = JSON.stringify(jsonFormatData);
        this.setState({ json: jsonData });
      })
      .catch(error => {
        console.error(error);
      });
  }

  render() {
    Survey.Survey.cssType = "bootstrap";
    this.model = new Survey.Model(this.state.json);
    this.model.data = this.state.answers;
    //set as read only
    this.model.mode = "display";

    return (
      <div>
        <div className="SurveyResult">
          <div className="surveyjs" >
            <Survey.Survey
              model={this.model}
            />

          </div>

        </div>
        {/* <FixedActions
              bgColor={this.state["color"]}
              bgImage={this.state["image"]}
              handleFixedClick={this.handleFixedClick}
              fixedClasses={this.state.fixedClasses}
          /> */}
      </div>
      
    );
  }
}

export default QuestionnaireResult;

