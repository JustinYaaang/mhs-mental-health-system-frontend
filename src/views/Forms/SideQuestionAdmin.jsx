import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// react plugin for creating charts
import ChartistGraph from "react-chartist";
// @material-ui/core
import withStyles from "@material-ui/core/styles/withStyles";
import Icon from "@material-ui/core/Icon";
// @material-ui/icons
import DateRange from "@material-ui/icons/DateRange";
import LocalOffer from "@material-ui/icons/LocalOffer";
import Update from "@material-ui/icons/Update";
import ArrowUpward from "@material-ui/icons/ArrowUpward";
import AccessTime from "@material-ui/icons/AccessTime";
import Accessibility from "@material-ui/icons/Accessibility";
import Grade from "@material-ui/icons/Grade";
import Code from "@material-ui/icons/Code";
import All from "@material-ui/icons/AllInboxOutlined";
// core components
import GridItem from "components/Grid/GridItem.jsx";
import GridContainer from "components/Grid/GridContainer.jsx";
import TaskView from "components/Tasks/TaskView.jsx";
import CustomTabs from "components/CustomTabs/CustomTabs.jsx";
import Card from "components/Card/Card.jsx";
import CardHeader from "components/Card/CardHeader.jsx";
import CardIcon from "components/Card/CardIcon.jsx";
import CardBody from "components/Card/CardBody.jsx";
import CardFooter from "components/Card/CardFooter.jsx";

import InformationCard from 'components/DashboardComponent/InformationCard.jsx';
import LineGraph from 'components/DashboardComponent/LineGraph.jsx';

import dashboardStyle from "assets/jss/material-dashboard-react/views/dashboardStyle.jsx";
import QuestionnaireAdmin from "../Questionnaire/QuestionnaireAdmin"

class SideQuestionAdmin extends React.Component{
    render(){
        return(
            <QuestionnaireAdmin question = {this.props.history} value = {12}/>
        )
    }
  }
  
  SideQuestionAdmin.propTypes = {
    classes: PropTypes.object.isRequired
  };
  
  export default withStyles(dashboardStyle)(SideQuestionAdmin);