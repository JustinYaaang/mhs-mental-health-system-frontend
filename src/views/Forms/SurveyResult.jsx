import React, { Component } from "react";
import * as Survey from "survey-react";
import "survey-react/survey.css";
import "bootstrap/dist/css/bootstrap.css";
import "jquery-ui/themes/base/all.css";
import "nouislider/distribute/nouislider.css";
import "select2/dist/css/select2.css";
import "bootstrap-slider/dist/css/bootstrap-slider.css";
import "jquery-bar-rating/dist/themes/css-stars.css";
import $ from "jquery";
import "jquery-ui/ui/widgets/datepicker.js";
import "select2/dist/js/select2.js";
import "jquery-bar-rating";
import * as widgets from "surveyjs-widgets";
import "icheck/skins/square/blue.css";
import axios from "axios";
import { getAnsweredQuestionnaire} from "../../services/BackendService";


import FixedActions from "components/FixedPlugin/FixedActions.jsx";

window["$"] = window["jQuery"] = $;
require("icheck");

Survey.StylesManager.applyTheme("darkblue");

widgets.icheck(Survey, $);
widgets.select2(Survey, $);
widgets.inputmask(Survey);
widgets.jquerybarrating(Survey, $);
widgets.jqueryuidatepicker(Survey, $);
widgets.nouislider(Survey);
widgets.select2tagbox(Survey, $);
widgets.signaturepad(Survey);
widgets.sortablejs(Survey);
widgets.ckeditor(Survey);
widgets.autocomplete(Survey, $);
widgets.bootstrapslider(Survey);

class SurveyResult extends Component {

  constructor(props) {
    super(props);
    this.state = {
      json:
      {
        "title": "",
        "description": "",
        "completedHtml": "",
        "pages": [],
        "showProgressBar": ""
      },
      answers: {}
    };
  }

  onValueChanged = (result) => {
    console.log("value changed!");
  }

  componentWillMount() {
    const {id} = this.props.match.params;
    console.log(id)
    getAnsweredQuestionnaire(id)
      .then(fetched_answers => {
        this.setState({ answers: JSON.parse(fetched_answers.body) });
        var jsonData = fetched_answers.questionnaireBody;
        var jsonFormatData = JSON.parse(jsonData);
        for (var i = 1; i < jsonFormatData.pages.length; i++) {
          if (jsonFormatData.pages[i].elements) {
            jsonFormatData.pages[0].elements = jsonFormatData.pages[0].elements.concat(jsonFormatData.pages[i].elements)
          }
        }
        jsonFormatData.pages = [jsonFormatData.pages[0]]
        jsonFormatData.pages[0].title = ""
        jsonData = JSON.stringify(jsonFormatData);
        this.setState({ json: jsonData });
      })
      .catch(error => {
        console.error(error);
      });
  }

  render() {
    Survey.Survey.cssType = "bootstrap";
    this.model = new Survey.Model(this.state.json);
    this.model.data = this.state.answers;
    //set as read only
    this.model.mode = "display";

    return (
      <div>
        <div className="SurveyResult">
          <div className="surveyjs" >
            <Survey.Survey
              model={this.model}
            />

            <center>
              <table border="1" width="180" >
                <tbody id="tbody1">

                </tbody>
              </table>
            </center>
            <div id="finalScore"></div>
            <div id="jsonSection"></div>
          </div>

        </div>
        <FixedActions
              //handleImageClick={this.handleImageClick}
              //handleColorClick={this.handleColorClick}
              bgColor={this.state["color"]}
              bgImage={this.state["image"]}
              handleFixedClick={this.handleFixedClick}
              fixedClasses={this.state.fixedClasses}
          />
      </div>
      
    );
  }
}

export default SurveyResult;

