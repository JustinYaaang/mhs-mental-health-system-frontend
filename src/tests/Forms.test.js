import React from 'react'
import { shallow, mount } from 'enzyme'
import ClinitianForm from '../views/Forms/ClinitianForm'
import PersonForm from '../views/Forms/PersonForm'
import TrustServiceForm from '../views/Forms/TrustServiceForm'
import UserDetail from '../views/Forms/UserDetail'
import SideQuestion from '../views/Forms/SideQuestion'
import SideQuestionAdmin from '../views/Forms/SideQuestionAdmin'
import SurveyResult from '../views/Forms/SurveyResult'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import QuestionnaireResult from 'views/Forms/QuestionnaireResult'
import QuestionnaireAdmin from 'views/Questionnaire/QuestionnaireAdmin'
configure({ adapter: new Adapter() })

describe('Clinitian Form', () => {
  it('should render correctly', async () => {
    const component = mount(<ClinitianForm />)

    expect(component).toMatchSnapshot()
  })
})
describe('Person Form', () => {
  it('should render correctly', async () => {
    const component = mount(<PersonForm />)

    expect(component).toMatchSnapshot()
  })
})

describe('Trust and Service Form', () => {
  it('should render correctly', async () => {
    const component = mount(<TrustServiceForm hasDetails={true} />)

    expect(component).toMatchSnapshot()
  })
})
describe('User details', () => {
  it('should render correctly', async () => {
    const component = mount(<UserDetail />)

    expect(component).toMatchSnapshot()
  })
})

describe('Side Question', () => {
  it('should render correctly', async () => {
    const component = mount(<SideQuestion />)

    expect(component).toMatchSnapshot()
  })
})
describe('Side Question Admin', () => {
  it('should render correctly', async () => {
    const component = mount(<SideQuestionAdmin />)

    expect(component).toMatchSnapshot()
  })
})
// describe('Survey Result', () => {
//   it('should render correctly', async () => {
//     const component = swallow(<SurveyResult />)

//     expect(component).toMatchSnapshot()
//   })
// })

describe('Questionnaire Result', () => {
  it('should render correctly', async () => {
    const component = mount(<QuestionnaireResult id="null" />)

    expect(component).toMatchSnapshot()
  })
})


describe('Questionnaire Admin', () => {
  it('should render correctly', async () => {
    const component = mount(<QuestionnaireAdmin />)

    expect(component).toMatchSnapshot()
  })
})