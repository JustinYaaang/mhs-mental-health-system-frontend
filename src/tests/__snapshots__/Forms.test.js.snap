// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clinitian Form should render correctly 1`] = `
<form
  class="trustform"
>
  <label
    className="label-subtitle"
    for="label"
  >
    Please enter the details of the NHS Person
  </label>
  <div
    class="form-group"
  >
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="firstnameinput"
      name="email"
      onChange={[Function]}
      placeholder="First Name"
      type="email"
    />
    <br />
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="lastnameinput"
      name="email"
      onChange={[Function]}
      placeholder="Last Name"
      type="email"
    />
    <br />
    <input
      id="STEP2"
      name="step"
      type="radio"
      value="STEP2"
    />
     Step 2 
    <br />
    <input
      id="STEP3"
      name="step"
      type="radio"
      value="STEP3"
    />
     Step 3
    <br />
  </div>
  <div
    class="form-group"
  >
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="emailinput"
      name="email"
      onChange={[Function]}
      placeholder="email"
      type="email"
    />
    <br />
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="passwordinput"
      name="email"
      onChange={[Function]}
      placeholder="password"
      readOnly={true}
      type="email"
    />
    <br />
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="trustnameinput_postcodeinput"
      name="email"
      onChange={[Function]}
      placeholder="Trust Name"
      readOnly={true}
      type="email"
    />
  </div>
  <br />
  <label
    className="label-subtitle"
    for="label"
    id="passwordlabel"
  >
    Enter password here:
  </label>
  <div>
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="passwordchange1"
      name="email"
      onChange={[Function]}
      placeholder="Enter password"
      type="email"
    />
    <br />
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="passwordchange2"
      name="email"
      onChange={[Function]}
      placeholder="Enter password again"
      type="email"
    />
  </div>
  <WithStyles(RegularButton)
    color="primary"
    id="loginbutton"
    onClick={[Function]}
    type="button"
  >
    Save
  </WithStyles(RegularButton)>
</form>
`;

exports[`Person Form should render correctly 1`] = `
<form
  class="trustform"
>
  <label
    className="label-subtitle"
    for="label"
  >
    Please enter the details of the NHS Person
  </label>
  <div
    class="form-group"
  >
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="firstnameinput"
      name="email"
      onChange={[Function]}
      placeholder="First Name"
      type="email"
    />
    <br />
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="lastnameinput"
      name="email"
      onChange={[Function]}
      placeholder="Last Name"
      type="email"
    />
  </div>
  <div
    class="form-group"
  >
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="emailinput"
      name="email"
      onChange={[Function]}
      placeholder="email"
      type="email"
    />
    <br />
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="passwordinput"
      name="email"
      onChange={[Function]}
      placeholder="password"
      readOnly={true}
      type="email"
    />
    <br />
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="trustnameinput_postcodeinput"
      name="email"
      onChange={[Function]}
      placeholder="Trust Name"
      readOnly={true}
      type="email"
    />
  </div>
  <br />
  <label
    className="label-subtitle"
    for="label"
    id="passwordlabel"
  >
    Enter password here:
  </label>
  <div>
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="passwordchange1"
      name="email"
      onChange={[Function]}
      placeholder="Enter password"
      type="email"
    />
    <br />
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="passwordchange2"
      name="email"
      onChange={[Function]}
      placeholder="Enter password again"
      type="email"
    />
  </div>
  <WithStyles(RegularButton)
    color="primary"
    id="loginbutton"
    onClick={[Function]}
    type="button"
  >
    Save
  </WithStyles(RegularButton)>
</form>
`;

exports[`Trust and Service Form should render correctly 1`] = `
<form
  class="trustform"
>
  <label
    className="label-subtitle"
    for="label"
    id="mainlabel"
  >
    Please enter the details of the NHS Trust.
  </label>
  <div
    class="form-group"
  >
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="nameinput"
      name="email"
      onChange={[Function]}
      placeholder="Name"
      type="email"
    />
  </div>
  <div
    class="form-group"
  >
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="address1input"
      name="email"
      onChange={[Function]}
      placeholder="Address Line 1"
      type="email"
    />
    <br />
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="address2input"
      name="email"
      onChange={[Function]}
      placeholder="Address Line 2"
      type="email"
    />
    <br />
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="postcodeinput"
      name="email"
      onChange={[Function]}
      placeholder="Postcode"
      type="email"
    />
  </div>
  <div
    class="form-group"
  >
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="descriptioninput"
      name="email"
      onChange={[Function]}
      placeholder="Description"
      type="email"
    />
    <br />
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="websiteinput"
      name="email"
      onChange={[Function]}
      placeholder="Website link"
      type="email"
    />
    <br />
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="emailinput"
      name="email"
      onChange={[Function]}
      placeholder="email"
      type="email"
    />
    <br />
    <input
      aria-describedby="emailHelp"
      class="form-control"
      id="telephoneinput"
      name="email"
      onChange={[Function]}
      placeholder="Telephone"
      type="email"
    />
  </div>
  <WithStyles(RegularButton)
    color="primary"
    id="loginbutton"
    onClick={[Function]}
    type="button"
  >
    Submit
  </WithStyles(RegularButton)>
</form>
`;

exports[`User details should render correctly 1`] = `
<UserDetail
  classes={
    Object {
      "cardCategoryWhite": "UserDetail-cardCategoryWhite-1",
      "cardTitleWhite": "UserDetail-cardTitleWhite-2",
    }
  }
/>
`;
